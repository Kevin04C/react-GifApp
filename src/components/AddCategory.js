import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setcategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);

    console.log("handleInputChange llamado",inputValue);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setcategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    } else {
      alert("No podes dejar el campo vacio");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
        id="input"
        autoComplete="off"
      />
    </form>
  );
};

AddCategory.propTypes = {
  setcategories: PropTypes.func.isRequired,
};
