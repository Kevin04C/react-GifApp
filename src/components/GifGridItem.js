import React from 'react';

export const GifGridItem = ({id,title, url}) => {

  return (
    <div className='gifs__item animate__animated animate__fadeInUpBig '>
      <img src={url} alt={title}/>
      <p> { title }</p>
    </div>
  )
}
