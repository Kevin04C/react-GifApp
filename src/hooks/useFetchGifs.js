import { useEffect, useState } from "react";
import {getGift} from '../helpers/getGift'

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    laoding: true,
  });

  useEffect(() => {
    getGift(category)
      .then((imgs) =>
        setState({
        data: imgs,
        laoding: false,
      })
    );
  }, [category]);

  return state;
};
