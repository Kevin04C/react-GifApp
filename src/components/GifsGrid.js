import React from "react";
import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "./hooks/useFetchGifs";
import laoder from '../assets/loader.svg'

export const GifsGrid = ({ category }) => {
  const { data: Images, laoding } = useFetchGifs(category);

  console.log(Images);

  return (
    <>
      <h3>{category.toUpperCase()}</h3>
      
      {laoding && <img src={laoder} alt="cargando" className="loader" />}

      <div className="gifs__container">
        {Images.map((image) => (
          <GifGridItem {...image} key={image.id} />
        ))}
      </div>
    </>
  );
};
