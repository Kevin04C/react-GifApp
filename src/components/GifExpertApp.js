import React, { useState } from "react";
import { AddCategory } from "./AddCategory";
import { GifsGrid } from "./GifsGrid";

export const GifExpertApp = () => {
  const [categories, setcategories] = useState();
  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setcategories={setcategories} />
      <hr />
      {categories.map((category) => (
        <GifsGrid category={category} key={category}/>
      ))}
    </>
  );
};
