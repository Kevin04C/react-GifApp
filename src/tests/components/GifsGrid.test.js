import React from "react";
import { shallow } from "enzyme";
import { GifsGrid } from "../../components/GifsGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("Prubea en el componente <GifsGrid />", () => {
  test("Debe devolver un snapshot del componente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      laoding: true,
    });

    const wrapper = shallow(<GifsGrid category="un show mas" />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar los items cuando se carga la informacion", () => {
    const imgs = [
      {
        id: "ABC",
        url: "https://locahost/cualquiercosa.jpg",
        title: "Cualquier cosa",
      },
      {
        id: "123",
        url: "https://locahost/cualquiercosa.jpg",
        title: "Cualquier cosa",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: imgs,
      laoding: true,
    });
    const wrapper = shallow(<GifsGrid category="un show mas" />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe(imgs.length);


  });
});
