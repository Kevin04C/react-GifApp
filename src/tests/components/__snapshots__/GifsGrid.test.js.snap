// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prubea en el componente <GifsGrid /> Debe devolver un snapshot del componente 1`] = `
<Fragment>
  <h3>
    UN SHOW MAS
  </h3>
  <img
    alt="cargando"
    className="loader"
    src="loader.svg"
  />
  <div
    className="gifs__container"
  />
</Fragment>
`;

exports[`Prubea en el componente <GifsGrid /> debe mostrar los items cuando se carga la informacion 1`] = `
<Fragment>
  <h3>
    UN SHOW MAS
  </h3>
  <img
    alt="cargando"
    className="loader"
    src="loader.svg"
  />
  <div
    className="gifs__container"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://locahost/cualquiercosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://locahost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
